### DISCUSSION TITLE
Unified Alerting: Option for "keep last state" or "don't send a notification" for a "No data" condition.

### DISCUSSION THREAD

**Discussion by Jessimon** [grafana/grafana#43588] [Alerting]
Hello Everyone,

The option to "keep last state" for No Data handling has been removed in Unified Alerting.
This unfortunately generates a lot of notifications especially during low use periods.

It also makes it hard when using custom templates for notifications.

Is it possible to get an extra option for "no data" to not send any notification or to keep the last state when this happens?

Many thanks

**Comment by anavss** [comment]
Hi @Jessimon 👋🏼 one of our product designers is investigating usability improvements to the new Unified Alerting released in Grafana 8.3. I spotted your request and wondered if you would be interested in talking to her? You can [book a time with us here](https://ethn.io/75322). 🙂

**Reply by Jessimon** [reply]
Hello Anavss,

I would love to but your invite link does not work.
https://ethn.io/....
ERR_SSL_KEY_USAGE_INCOMPATIBLE

Kind regards,
Jessimon

**Reply by anavss** [reply]
Oh sorry about that. Here it goes again: https://ethn.io/75322. If it still doesn't work, [email me please](mailto:ana.vanschalkwyk@grafana.com) and we can sort it out.

**Comment by grobinson-grafana** [comment]
Hi @Jessimon,

When **No Data** is selected for **Alert state if no data or all values are null** the alert behaves as if it was set to keep last state and a second alert is fired: **DatasourceNoData**. However, it's possible to silence **DatasourceNoData** alerts to stop receiving notifications for these alerts via the Silences tab.

**Reply by Jessimon** [reply]
Hello @gerobinson,

You are correct, this works for the firing notification.
Unfortunately it does not work for the resolved notification.
In a resolved notification alertname is not equal to "DatasourceNoData" and so the silence does not trigger.
And so the receiver of the notification get a resolved message with the name of the alert that could be true or false.
Also if the alert query goes from no data to firing you get first a resolved message with the name of the alert and then an firing notification after a while (depending on alarm settings).
This is very confusing.

**Reply by Jessimon** [reply]
Rectification.
There are now correct DatasourceNoData resolved notifications.
[RESOLVED] (DatasourceNoData ...
There where somewhat buried in all the NaN alerts.

**Comment by jgournet** [comment]
@grobinson-grafana or @anavss :
Would you know if this would be possible to implement a "proper" setting for this ?
ie: add a "keep alert state" in this dropdown:
![image](https://user-images.githubusercontent.com/54783141/232974033-bc58606e-a56b-472f-8492-01adf79ca69e.png)

This is because using a "global" way of silencing alerts does not seem right - it feels very error prone and could easily silence alerts we don't want silenced.

references:
https://github.com/grafana/grafana/discussions/55899
https://github.com/grafana/grafana/issues/33828

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
